core/src/main/java/org/apache/carbondata/core/scan/filter/FilterUtil.java

```diff
@@ -188,6 +189,14 @@ private static FilterExecuter createFilterExecuterTree(
           return new FalseFilterExecutor();
         case ROWLEVEL:
         default:
+          if (filterExpressionResolverTree.getFilterExpression() instanceof UnknownExpression) {
```

Comment:
can we add new  expression type and filter executor type?

Reply:
In the initial version of the feature PR, polygon expression and executor type were defined explitily for polygon filter in carbondata-core module itself. But Ravi's review comment suggested the use of `UnknownExpression`(similar to `SparkUnknownExpression`) and have carbondata-geo depends on carbondata-core. But not the other way round.